ceph.spec.in

```diff
@@ -201,18 +201,24 @@ BuildRequires:  librdkafka-devel
 %if 0%{with make_check}
 BuildRequires:  jq
 BuildRequires:	libuuid-devel
-BuildRequires:	python%{python3_pkgversion}-bcrypt
+BuildRequires:	python%{python3_version_nodots}-bcrypt
+%if 0%{?rhel} == 8
 BuildRequires:	python%{python3_pkgversion}-nose
-BuildRequires:	python%{python3_pkgversion}-requests
 BuildRequires:	python%{python3_pkgversion}-six
 BuildRequires:	python%{python3_pkgversion}-virtualenv
+%else
+BuildRequires:	python%{python3_version_nodots}-nose
+BuildRequires:	python%{python3_version_nodots}-six
+BuildRequires:	python%{python3_version_nodots}-virtualenv
+%endif
+BuildRequires:	python-tox
+BuildRequires:	python%{python3_pkgversion}-requests
 %if 0%{?rhel} < 8
-BuildRequires:	python%{python3_pkgversion}-coverage
-BuildRequires:	python%{python3_pkgversion}-pecan
-BuildRequires:	python%{python3_pkgversion}-tox
+BuildRequires:	python%{python3_version_nodots}-coverage
+BuildRequires:	python-pecan
```

Comment:
we've dropped python2 support.

Reply:
as you can see, there is no `python3-pecan` related package in CentOS7:

```bash
$ yum search python pecan
python-pecan.noarch: A lean WSGI object-dispatching web framework
python2-pecan.noarch: A lean WSGI object-dispatching web framework
```
`python3-routes`, `python3-tox`, `python3-werkzeug` and `python3-cherrypy` are also not available in CentOS7. do you have any suggestions?

Reply:
i think we can leave them as they are. as these packages are not likely to be packaged for el7. and they are not build dependencies for packaging ceph, so there is no point to conditionalize for them.

Reply:
`install-dep.sh` does not completely install all the dependencies required by the ceph development environment (eg mgr: dashboard / mgr: restful). The SPEC file is mainly used to package ceph. May I think so?

Reply:
td;dr. in el7, yes.

ceph.spec.in is used for preparing the build dependencies for both packaging and testing. the latter is enabled by the option "make_check", which is in turn enabled by `install-deps.sh`. we expect developers to migrate to newer distros like el8 and bionic for hacking ceph.

the reason why el7 matters is that i believe we can still have a minimal set of ceph packages that can help users who need to stick with el7. and i assume that they can live with ceph without dashboard which depends on the missing dependencies you listed above, like `python3-routes`.

Reply:
you could comment out the following block
```sh
if tty -s; then
    # interactive
    for_make_check=true
elif [ $FOR_MAKE_CHECK ]; then
    for_make_check=true
else
    for_make_check=false
fi
```
in `install-deps.sh` for testing your change.